api/v1alpha1/databasecluster_types.go

```diff
@@ -310,6 +311,14 @@ type Monitoring struct {
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 }
 
+// Sharding are the sharding options. Available only for psmdb
+type Sharding struct {
+	Enabled      bool                 `json:"enabled"`
+	Shards       int32                `json:"shards"`
```

Comment:
The number will be mapped to the psmdb `.Spec.Replsets` section so that a cluster will have as many replsets as the number of shards `N`. 
The list of the replsets by default contains only one replset `rs0`, but with the sharding is enabled the list will contain N (probably identical) replsets `rs0`, `rs1`, ... `rs(N-1)`